eogrow/utils/grid.py

```diff
@@ -95,3 +95,35 @@ def get_enclosing_bbox(bboxes: Sequence[BBox]) -> BBox:
 
     collection = GeometryCollection([bbox.geometry for bbox in bboxes])
     return BBox(collection.bounds, crs=bboxes[0].crs)
+
+
+def split_bbox(
+    named_bbox: Tuple[str, BBox],
+    split_x: int,
+    split_y: int,
+    buffer_x: float,
+    buffer_y: float,
+    naming_schema: str = "{name}_{i_x}_{i_y}",
+) -> List[Tuple[str, BBox]]:
+    """Splits a BBox into multiple smaller BBoxes with new names generated for them.
+
+    The `buffer` parameters describe the buffer of the original BBox, which is copied to the split ones.
+    """
+    name, bbox = named_bbox
+    min_x, min_y = bbox.lower_left
+    max_x, max_y = bbox.upper_right
+
+    ll_xs, x_step = np.linspace(min_x + buffer_x, max_x - buffer_x, split_x, endpoint=False, retstep=True)
+    ll_ys, y_step = np.linspace(min_y + buffer_y, max_y - buffer_y, split_y, endpoint=False, retstep=True)
```

Comment:
`ll_xs` and `ll_ys` not descriptive enough.

Reply:
Would it suffice if i add a comment `# the ll_ prefix stands for lower_left`?

Reply:
Perhaps there is no need to use `ll` in the first place and would be more intuitive with:

`x_edges, x_step = ...`